### DISCUSSION TITLE
Let's Shrink LBB - and make it Portable

### DISCUSSION THREAD

**Discussion by ruebyi** [outdoorbits/little-backup-box#187] [Ideas]
Hi, 
is here anyone who would be interested in a shrinker Version of LBB?

I'm thinking about a Pi Zero 2 with Hub and Batterypack in a small 3D- Printed Case with LCD.

With MicroSDs up to 1TB we shouldn't need a SSD or HDD.
I could buy the pi and test the software, but I'm not able to design a Case...
Perhaps someone already has made up his mind on a battery driven LBB?

Would be glad to find others to bring this idea to life!

**Comment by outdoorbits** [comment]
That sounds like a great idea! I tried to design a case for the Pi4, optimised to my display but failed in lack of time.
It would be nice, if the drawings could be done on an open source and platform independent cad software like freecad.

**Comment by rkosick** [comment]
I build something similar using a Pi Zero 2W and a ZERO4U_2018_801 Hub, but without a battery pack. I described it here:

https://discuss.pixls.us/t/case-for-raspberry-pi-zero-2-w-plus-zero4u-and-0-96-inch-display/28326

The Pi Zero 2W works fine. File transfer reaches the limits of USB2.
Keep in mind it’s not the most beautiful case because it was my first try with a CAD Software. Always wanted to modify it. But I'm not sure about the battery pack. It would make it larger again and I always have one with me anyway.

**Reply by ruebyi** [reply]
Great! Thank you!
Your case could be a good start for me. 
rught now I’m struggling to get a pi zero2 w. 
it‘s sold out everywhere.

**Reply by rkosick** [reply]
Yeah it's still hard to get anything for a good price. If you want to try the case don't miss the latest comment in the thread. There is the latest version of it.

**Comment by SigmaFred** [comment]
@ruebyi @rkosick 
I built something similar too. Based on a  Pi Zero 2W, Waveshare USB Hub with Pogo-Pins plus 1.3" OLED Display HAT für Raspberry Pi. The result is compact and with the 3D-printed case a slim, quite robust and portable solution. I use it with an external battery pack on the road to backup pictures from a camera SD card to tiny USB sticks (for redundancy and loss-prevention). See pictures below

If of interest - I am happy to share the 3D Files (STL, STEP, Fusion360) of the case and more details on used parts. The setup of the buttons and display is equal to the version of "roshin8"-> [link](https://github.com/outdoorbits/little-backup-box/discussions/242)

<img src="https://github.com/outdoorbits/little-backup-box/assets/23003780/d82a74e0-14fa-4917-8aa2-b6f3abe897a6" width="90%" height="90%"/>
<img src="https://github.com/outdoorbits/little-backup-box/assets/23003780/4dcf5efe-bdea-4a4e-83df-03521b2c8d44" width="45%" height="45%"/>
<img src="https://github.com/outdoorbits/little-backup-box/assets/23003780/cefeecb3-fdfc-4060-90bb-47e8bbf895da" width="45%" height="45%"/>
<img src="https://github.com/outdoorbits/little-backup-box/assets/23003780/16f9a3b1-df86-47b3-a023-b32ccd2fa7b6" width="45%" height="45%"/>

Edit: Resized pictures a bit (still clickable for full size)

**Comment by ruebyi** [comment]
Great!

would be nice if we could hear some more!

**Comment by outdoorbits** [comment]
This is a very nice solution. If you wouldn't mind, I'd like to put your photos into the gallery. I also could put the STL files into the case repository.

**Reply by SigmaFred** [reply]
Would love to see the pictures and files in the repository ... AS SOON AS  I have redesigned the case for the new USB Hub.

As I faced regular issues on the road with this setup due to WiFi interference of the Zero 2 W with the Waveshare USB Hub, I will replace the Waveshare USB Hub with Zero4U (from UUgear) as they seem to have managed the issue with use of a ferrit ring and a metallised PET membrane. I already tested it and it is much more stable than the Waveshare version. Unfortunately using Zero4U will require some changes/adjustments on the case, as the USB slots have different positions and the board is slightly higher. 

Underlying complication*:"After complete tests we realize that Raspberry Pi Zero 2 W has stronger WiFi radiation than the first generation of Raspberry Pi Zero W. As a result, the USB devices connected to some ports in Zero4U may get interfered and work unstably. "
*) quoted from uugear

One question in this context: as I do not need WiFi at all when backing up "on the road" -would there be a chance to add an option in the menu of LBB to disable / enable WiFi in general on the Raspi or an option to "calm down" comitup WiFi usage e.g. to just offer the hotspot (refrain from trying / "connecting ...") - as anything that reduces WiFi activities would reduce interference and help me improve the stability. 

I am sure this complication is just due to the compact built, and I know I could easily circumvent it by using a regular sized raspi, but I would love to keep LBB at that tiny size. Thx.

**Comment by outdoorbits** [comment]
Disabling WiFi would be a one-way street, lbb would then no longer be 
accessible or configurable. That wouldn't exactly be my favorite.
However, to reduce Comitup to just a hotspot, you could simply delete 
its configuration. This can be done via the web UI: The access data can 
be deleted under Settings - Comitup.

I look forward to your update!

**Reply by SigmaFred** [reply]
UPDATE:

New Hardware Setup: 
- Pi Zero 2W, 
- Zero4U USB Hub with Pogo-Pins (UUGear)
- 1.3" OLED Display HAT for Raspberry Pi 128x64 (Waveshare) 

Created a case with a protective cover to be able to put it as is into my camera bag. 

<img src="https://github.com/outdoorbits/little-backup-box/assets/23003780/8ffba26c-ad00-46b3-bca3-4c6d54db30d2" width="45%" height="45%"/>
<img src="https://github.com/outdoorbits/little-backup-box/assets/23003780/bb4d69b6-1d70-4df7-9317-16a2d6d1717e" width="45%" height="45%"/>
<img src="https://github.com/outdoorbits/little-backup-box/assets/23003780/fc8be919-e15d-4748-8624-6f9ddabbfafa" width="45%" height="45%"/>
<img src="https://github.com/outdoorbits/little-backup-box/assets/23003780/1da650de-e66f-4cab-a765-8521dc8c6d34" width="45%" height="45%"/>
<img src="https://github.com/outdoorbits/little-backup-box/assets/23003780/98878a94-a243-4842-ac74-80f68b921722" width="45%" height="45%"/>
<img src="https://github.com/outdoorbits/little-backup-box/assets/23003780/e8fd3c78-a9df-4af7-bc08-98a59ad28d69" width="45%" height="45%"/>
<img src="https://github.com/outdoorbits/little-backup-box/assets/23003780/915a44e4-bd56-4e85-a051-65f5ce495b4d" width="45%" height="45%"/>
<img src="" width="45%" height="45%"/>
<img src="https://github.com/outdoorbits/little-backup-box/assets/23003780/0236fbfd-b902-4975-aafe-14c1cb52be46" width="45%" height="45%"/>
<img src="https://github.com/outdoorbits/little-backup-box/assets/23003780/fc7128e7-9e70-4a6b-a5ad-7f41b326a50f" width="45%" height="45%"/>

Link to STL files -> https://github.com/outdoorbits/case-for-little-backup-box/tree/main/Raspberry_Pi_zero_2W
(now stored in the github repository)

Edit: Changed links to STL and resized pictures a bit (still clickable for full size)

**Reply by DSew20** [reply]
This does look like a great little solution.
If the zero 2w fast enough for the job? Any news on sharing the case files? Would like to try a build too.

**Reply by SigmaFred** [reply]
> This does look like a great little solution. If the zero 2w fast enough for the job? Any news on sharing the case files? Would like to try a build too.

1. For me it is fast enough - in my setup (see above) 400 pics take less than 10 min to copy. 
2. I added an interim link to the STL files above.

**Reply by DSew20** [reply]
Thank you. Do you know what size standoffs you needed to get this to stack and fit in the case?

**Reply by SigmaFred** [reply]
All M2.5 ... 
<img src="https://github.com/outdoorbits/little-backup-box/assets/23003780/1e9aac7c-7e71-4dd5-aac2-e040d995596f" width="45%" height="45%"/>

**Reply by sejtam** [reply]
https://github.com/outdoorbits/case-for-little-backup-box/tree/main/Raspberry_Pi_zero_2W

I had the STL files for the ZeroW2 (no battery) case printed at a 3d printing service, and I find that the cover and main case are too tight. It's very hard to get them apart again. The really tony tab on one side does not help. I think this would require some additional handles to get a good grip for pulling them apart

**Reply by SigmaFred** [reply]
Sorry to hear about this problem. It perfectly slides in/out with my own printout. To help for your case I created a version which has a slightly wider "gap" between inner and outer box. I hope this helps.
@outdoorbits - maybe you can add this version to the repository too. The original version has a 0.05mm gap and prints perfectly fine with my Bambulab P1S. The additional version has a 0.1mm gap and may be an option when printout is too tight.
[RaspiZero2W_gap_0.1mm.zip](https://github.com/user-attachments/files/21066134/RaspiZero2W_gap_0.1mm.zip)

**Reply by outdoorbits** [reply]
Thank you very much, the new variant is published here: [https://github.com/outdoorbits/case-for-little-backup-box/tree/main/Raspberry_Pi_zero_2W/Tiny_case](https://github.com/outdoorbits/case-for-little-backup-box/tree/main/Raspberry_Pi_zero_2W/Tiny_case)

**Reply by sejtam** [reply]
Thanks for that. I really appreciate it.  I have made the older one work for now with copious sanding of bot the inner surface of the cover and the outer of the main case, but I will try the new version when I have the next one printed.  (MY son's own 3d printer is packed away, so I use a service for now)

**Comment by ruebyi** [comment]
You are a genius!
Have you ever thought on adding a battery?

**Reply by SigmaFred** [reply]
Thx for the flowers :-) 
For me an integrated battery was not a requirement so far - so I did not seek/look for those components. 
Do you have a suggestion for a certain battery pack on the market that you find most promising for such a setup?

**Reply by ruebyi** [reply]
I thought about something like this:
Geekworm Ultra-thin 18650 UPS & USB HUB Shield (X306) for Raspberry Pi Zero 2 W / Zero W https://amzn.eu/d/bEtu7Jq

**Reply by SigmaFred** [reply]
Did you setup/test LBB with this shield already?  Is a single "18650" battery sufficient for the intended usage?
For me it would be an interesting setup, if I could survive about 1 week without recharging (= 7x2 backups each about 5-7 minutes) which means the battery should run the raspi about 2 h. Any estimates?

Purely looking at the hard-/software - I think stacking (from top to bottom)
- 1.3" OLED Display HAT for Raspberry Pi 128x64 (Waveshare)
- Pi Zero 2W,
- Geekworm Ultra-thin 18650 UPS & USB HUB Shield (X306) for Raspberry Pi Zero 2 W / Zero W

should work with the same configuration as it does with my small solution.

Based on the raw outline of the Geekworm (64.5 x 77 mm) and the diameter of the battery (18.5mm) plus assuming 8 mm board height we could end up with a case 68 x 80 x 30mm, (if the piled OLED and Pi do not tower above the battery)

I think creating a case for this should not be too complicated - as soon as you have the exact dimensions and positions of all relevant components or better you have a sample to play with.

Edit: Found a 3D model for a case which hosts a geekworm plus Raspi Zero (but different display) ... which might help as starting point and size reference ... https://www.printables.com/de/model/454731-case-for-geekworm-x306-ups-shield-18650-battery-rp (size: 68x83x28.5 mm with 2mm walls)

Edit2: Bought the respective Geekworm shield (V1.2) and stacked the 3 boards as mentioned above - it works like a charm. Copy of 400 pic took less than 10 min. I think I will design a box for this setup as well THOUGH I think it might become hard in the near future to get such a geekworm shield with 3 USB ports: On the latest revision (V1.3) they omit those USB ports. Which means: Anyone interested in such a battery setup may need to be quick.

With this setup, the case mentioned above can be used as an interim solution, as the stacked boards fit in.

<img src="https://github.com/outdoorbits/little-backup-box/assets/23003780/579ec86a-98f4-4357-8ed6-8cacfd9b7211" width="45%" height="45%"/>
<img src="https://github.com/outdoorbits/little-backup-box/assets/23003780/260bf838-5278-44c2-8310-f10a7018ff8b" width="45%" height="45%"/>

<img src="https://github.com/outdoorbits/little-backup-box/assets/23003780/630a94c0-7dea-40c7-80ff-6905278a6bb6" width="45%" height="45%"/>

**Reply by ruebyi** [reply]
I‘m ordering therapist right now! Thanks for testing!

**Reply by SigmaFred** [reply]
### Pictures and details of new LBB case for this battery setup
-> with cover - overall 83.5x71.5x28 mm - and my copy weighs 161g 
-> case printed from PLA

<img src="https://github.com/outdoorbits/little-backup-box/assets/23003780/04f3059c-2c6a-478f-968d-1fb624ca493f" width="45%" height="45%"/>
<img src="https://github.com/outdoorbits/little-backup-box/assets/23003780/eb9f2827-7744-460d-8e49-1dcbe7b6f928" width="45%" height="45%"/>
<img src="https://github.com/outdoorbits/little-backup-box/assets/23003780/9d491234-42ea-4279-be1c-dd0cb3be3846" width="45%" height="45%"/>

Used components:
- 1.3" OLED Display HAT for Raspberry Pi 128x64 (Waveshare)
- Pi Zero 2W,
- Geekworm Ultra-thin 18650 UPS & USB HUB Shield (X306) for Raspberry Pi Zero 2 W / Zero W
- Small parts (see picture)

STL files can be found here -> https://github.com/outdoorbits/case-for-little-backup-box/tree/056f45a3579c40a3678385b22edd6b9d844b6396/Raspberry_Pi_zero_2W/Battery_case/stl

**Reply by ruebyi** [reply]
Looks awesome

**Reply by ruebyi** [reply]
I ran into problems today, 
It works without display and battery, but with display and battery it doesn’t boot.

I will test this,issue tomorrow furthermore

**Reply by SigmaFred** [reply]
Please test first with display only, as this combination is "passive" and also does not rely on "pogo-pins".

**Reply by ruebyi** [reply]
Works now with Display - only Comitup is forgetting my wifi all the time

**Reply by SigmaFred** [reply]
Are you shutting down LBB via menu and wait long enough? I had sometimes issues with my Raspi remembering the latest settings when I was too impatient.

**Reply by ruebyi** [reply]
Thank you! It works now! I did a complete reinstall and tried everything step by step! 
I will print the case next week and then it will be my perfect companion for Weddings and Wildlife for a backup on the go without Power!

**Comment by outdoorbits** [comment]
This looks really fantastic, congratulations!!!
If you agree, I would extend https://github.com/outdoorbits/case-for-little-backup-box by this as an own folder and rework the README.md.
This folder at case-for-little-backup-box would also be a good place for the stl files. I would also like to integrate your images in little-backup-box.

**Reply by SigmaFred** [reply]
I like the idea. Please proceed - and let me know how I can upload the respective STL files.

**Comment by outdoorbits** [comment]
Great! There are two ways: 

1. The GitHub way is to fork the repository, make your changes and a pull request. 

2. Just send the files to my mail address. You find it in the header of nearly every file of lbb.

Thank you!

**Reply by SigmaFred** [reply]
Tried to send an email with the stl files to your github@... address and got a mail delivery error 4 days later. Tried again without attachments - not sure if that worked. Pragmatically I now added an (interim) cloud storage link above in my update which points to these files. I hope this is OK. Will remove it as soon as we have the files in this lbb repository.

**Reply by SigmaFred** [reply]
For the second LBB-case I now tried the approach "fork and pull request".
=> https://github.com/outdoorbits/case-for-little-backup-box
There is a pull request open ... did you see that? 
Is there more to do from my side?

**Comment by outdoorbits** [comment]
I didn't recognise any mail problems but I'll check that. Anyway, I will publish your files and create a portal page. Here I hope for your corrections and contributions.

**Comment by outdoorbits** [comment]
A first version is uploaded. Please have a look at: https://github.com/outdoorbits/case-for-little-backup-box/tree/main/Raspberry_Pi_zero_2W
It's not linked yet and it's not yet represented into the gallery. I would like to complete the documentation before.
Did the joystick work out of the box or did you configure a special key assignment?

**Reply by SigmaFred** [reply]
joystick is setup as described by "roshin8"-> https://github.com/outdoorbits/little-backup-box/discussions/242 see his picture ... 
<img src="https://github.com/outdoorbits/little-backup-box/assets/23003780/f84276b7-f62b-4033-b872-81580df5a43c" width="45%" height="45%"/>
and please use the picture below  for "in action" as the one you placed there is the old version

<img src="https://github.com/outdoorbits/little-backup-box/assets/23003780/8ffba26c-ad00-46b3-bca3-4c6d54db30d2" width="45%" height="45%"/>

and "under the case" should be this

<img src="https://github.com/outdoorbits/little-backup-box/assets/23003780/915a44e4-bd56-4e85-a051-65f5ce495b4d" width="45%" height="45%"/>

**Reply by outdoorbits** [reply]
Ah, great, I'll update this.

**Reply by SigmaFred** [reply]
I did a little clean up and hope this collection suites better for the documentation and gives the needed info on how to build it up

<img src="https://github.com/outdoorbits/little-backup-box/assets/23003780/6716b079-7f33-4511-8f96-b54876d2ba05" width="45%" height="45%"/>
<img src="https://github.com/outdoorbits/little-backup-box/assets/23003780/8d130973-f262-40ae-830c-f6a71cd44b35" width="45%" height="45%"/>

**Comment by outdoorbits** [comment]
Great images. I replaced the most of the others and created links from lbb. Would you please check https://github.com/outdoorbits/case-for-little-backup-box/tree/main/Raspberry_Pi_zero_2W again? Any changes or errors?
Could you provide the CAD file(s) also? Which CAD software did you use?

**Reply by SigmaFred** [reply]
Designed in Autodesk Fusion (free - non commercial version) ...
Added file in cloud storage (.f3d)

**Comment by tutle** [comment]
Hello, great building ! Do you have any file transfer rate ? to compare with a pi5 config.
I want to build one for movie shooting so I need best rate as possible.
Sincerely

**Reply by SigmaFred** [reply]
I assume that for your intended use a Raspi 5 cannot be beaten when you compare the specifications:
With a Raspi Zero 2W you will have the limitations of USB2 while Raspi 5 supports USB3
USB Speed: USB 2.0 max 480 Mbps (Megabit/s) vs. USB 3.0 max 5 Gbps (Gigabit/s) 
USB Power: USB 2.0 max 500 mA vs. USB 3.0 max  900 mA
If you are fine not having a bigger screen the "1.3" OLED Display HAT for Raspberry Pi 128x64 (Waveshare)" should work also with the Pi5 (not tested). This may allow for a quite slim and robust case for travelling as well.

**Comment by outdoorbits** [comment]
About the screen it's the same driver. I have no doubt, it's working on 
both.

**Comment by outdoorbits** [comment]
No, thank you. I merged it. But the first version is duplicated now. Could you maybe adapt the readme of the zero2 folder as overview page?

**Reply by SigmaFred** [reply]
Changed as requested - "pull request" is open ... Thanks.

**Comment by outdoorbits** [comment]
To keep the content consistent, I just will change the links to 
outdoorbits, else they point to your fork.

**Reply by SigmaFred** [reply]
Can I avoid such extra work for you?
e.g. (how) can I use picture links and links to subdirectories that get automatically converted with a merge?
Sorry I am a newbee to GitHub usage

**Comment by outdoorbits** [comment]
Don't worry, I'm very happy about your contributions and it's not a big 
deal to change the links. You can't test your files without using links 
to your page.

**Comment by ruebyi** [comment]
So I've been on Vacation with this Little Backup Box:

Waveshare Display
Battery
USB HAT
3D printed Case
Raspberry Pi Zero 2 W

It worked perfectly! Backups were reliable and as fast as expected - 20 MB/s
That was enough for me - I put my SD Card in before Dinner and it was ready when I came back!

Rendering Previews took longer... much loooonger!
It took about 36h for 2000 pics

What's the Consequence for me?
I love my little LBB - as long as I Don't need the previews!
I will create an LBB with a pi 5 - also with Battery, NVME and Waveshare Display to have a second LBB with more performance!

**Comment by Physikant85** [comment]
Hi, 
I tried to reproduce your setup! Bought a little bit different USB-hub, but the display is exactly the same. Unfortunately, I can not get the display to work.

It is factory-configured as 4-Wire-Spi with CS0 (based on SH1106).

I configured it that way and checked the mark "use display" in the display section. After this first config, it (so the display) showed "Einstellung gespeichert" (so settings saved, I use the german version) in the display. But since then, nothing can be seen anymore, even after a power cycle, the display stays black. Even saving the config again doesen't trigger a reaction from the display. Any idea what could go wrong here?

I could change the jumpers to 3-wire-SPI or I2C, but I don't see how this changes anything since the display already showed text, but only once, so the communication is there ... 

In the device status section (via remote setup) i get the following message:
ERROR: smartctl requires a device name as the final command-line argument. 				  				 					
 Is it related? And if I want to download the config, the .zip file is empty...

Any ideas what I'm doing wrong?

**Reply by SigmaFred** [reply]
Hi,
sorry about this. As it worked straight away in my setup, I have no clue what goes wrong in your case. Yust as a first cycle of trouble shooting: Did you check if the display works with the Raspi + screen only (omitting USB-Hub)? Just to eliminate a component that might interfere with screen/raspi in this setup.

**Reply by outdoorbits** [reply]
As it worked one time and after that it didn't without any change, I would hunt for hardware problems and keep the software setup as it was. 
Are there connection problems - are the solderings and the cables ok?
Did you connect to the correct power pin (5V vs. 3.3V)?

**Reply by Physikant85** [reply]
Hi guys,
thanks for the answers so far!
Im really happy someone tries to help me :)
![disp](https://github.com/user-attachments/assets/db230c2a-49fa-45ab-b621-12b4bece3fb9)
![mounted](https://github.com/user-attachments/assets/6f53f797-4fe6-40c3-81be-1ec216c9a70d)
![PiZero2W](https://github.com/user-attachments/assets/b80f05ed-5899-43a4-a550-1edb801fb40c)
![USBHub](https://github.com/user-attachments/assets/25ac8938-91f6-4925-aa56-5cd43c4f2293)
Here you can see my setup. I tried running it without the USB hub, but that doesen't change anything :(
![Screenshot_20241206_121445_Firefox](https://github.com/user-attachments/assets/cdb29f68-ccb5-466d-91e8-d7f5ba54c788)
![Screenshot_20241206_121511_Firefox](https://github.com/user-attachments/assets/89f3ff40-b7ff-406f-9d60-bf62da75d0ee)
![Screenshot_20241206_121520_Firefox](https://github.com/user-attachments/assets/38e913b3-238a-4a63-9f2e-7fe302749fcb)
![Screenshot_20241206_121525_Firefox](https://github.com/user-attachments/assets/99c31847-5215-4c9c-828c-08e7f46a7e64)
![Screenshot_20241206_121533_Firefox](https://github.com/user-attachments/assets/ffabd42d-6eb2-4973-839d-67645fcd0313)
![Screenshot_20241206_121351_Firefox](https://github.com/user-attachments/assets/2a43fd55-cf6e-4858-a092-4111b75d3223)
![Screenshot_20241206_121436_Firefox](https://github.com/user-attachments/assets/99c9b55d-1a28-4e00-b8c9-c7b9989f1127)
Here you can see my configuration.
Does the 
"ERROR: smartctl requires a device name as the final command-line argument." have any meaning?

**Reply by outdoorbits** [reply]
The smartctl error is associated to a storage device, not to the display.

**Reply by outdoorbits** [reply]
By the way: This error message should not come up anyway, it's fixed now.

**Reply by outdoorbits** [reply]
Why is contrast reduced? Did you try full contrast?
